hack/v1migration/phase1.sh

```diff
@@ -0,0 +1,50 @@
+#!/bin/sh
+
+set -e
+
+if [ $# != 1 ]; then
+	echo "USAGE: $0 [WORKDIR]"
+	exit 1
+fi
+
+WORKDIR=$1
+echo "Using workdir: $WORKDIR"
+
+oc project hive
+
+oc scale -n hive deployment.v1.apps/hive-operator --replicas=0
+oc scale -n hive deployment.v1.apps/hive-controllers --replicas=0
+oc scale -n hive deployment.v1.apps/hiveadmission --replicas=0
+
+# TODO: wait until no pods are running
+
+mkdir -p $WORKDIR/
+
+HIVE_TYPES=( checkpoints clusterdeployments clusterdeprovisionrequests clusterimagesets clusterprovisions clusterstates dnsendpoints dnszones hiveconfig selectorsyncidentityprovider selectorsyncset syncidentityprovider syncsetinstance syncset )
+for i in "${HIVE_TYPES[@]}"
+do
+	:
+	echo "Storing all ${i} in ${WORKDIR}/${i}.yaml"
+	oc get ${i}.hive.openshift.io -A -o json | jq '.items | .[] |
+		del(.status) |
+		del(.metadata.annotations."kubectl.kubernetes.io/last-applied-configuration") |
+		del(.metadata.creationTimestamp) |
+		del(.metadata.generation) |
+		del(.metadata.resourceVersion) |
+		del(.metadata.ownerReferences) |
```

Comment:
I would expect us to keep the owner references in the yaml so that we can use them to re-apply the owner references on the restored resources.

Reply:
My understanding was we were going to rollout @twiest 's work and let that use labels to restore owner refs.

Reply:
That will require us to apply the labels to the v1alpha1 objects. It doesn't strike me as too much harder to restore the owner references directly than it is to apply the labels. And then we do not *need* to have Thomas's work.